fili-core/src/main/java/com/yahoo/bard/webservice/druid/serializers/Util.java

```diff
@@ -24,24 +25,42 @@
      * @return  an Optional String of physical name
      */
     public static Optional<String> findPhysicalName(Dimension value, JsonGenerator gen) {
-        JsonStreamContext context = gen.getOutputContext();
         String apiName = value.getApiName();
         // Search for physical name
+        return withDruidQuery(gen, druidQuery -> druidQuery.getDataSource().getPhysicalTables().iterator().next().getPhysicalColumnName(apiName)
+        );
+    }
+
+    /**
+     * JSON tree walk to determine if there is a nested query below the current json node or not.
+     *
+     * @param gen  the Json Generator to retrieve the tree to walk on.
+     *
+     * @return  a Boolean where true indicates there are more nested query below this node, false otherwise
+     */
+    public static Boolean hasInnerQuery(JsonGenerator gen) {
+        return withDruidQuery(gen, druidQuery -> druidQuery.getInnerQuery() != null).orElse(false);
+    }
+
+    /**
+     * JSON tree walk to find the druid query context of the current context,
+     * returns the current context if current context is a druid query.
+     *
+     * @param gen  the Json Generator to retrieve the tree to walk on.
+     * @param success  a function that takes an DruidQuery as an argument and return the final desired returned result.
+     *
+     * @return an Optional of DruidQuery if found, Optional.empty otherwise
+     */
+    public static <T> Optional<T> withDruidQuery(JsonGenerator gen, Function<DruidQuery, T> success) {
+        JsonStreamContext context = gen.getOutputContext();
```

Comment:
the idea behind `with` in this case is that it's executing the function _in the context of_ (or _with_) the `DruidQuery`. Groovy has a very similar construct, which is what this method is (very roughly) mimicking:

```groovy
// If we have an object with methods:
class Foo {
    def doBar() { ... }
    def doBaz() { ... }
}
def myFoo = new Foo()

// This:
myFoo.with {
    doBar()
    doBaz()
}

// is the same as:
myFoo.doBar()
myFoo.doBaz()
```

The previous name for this method in this PR was `getDruidQuery`. Thus far, we don't really have this pattern in the Fili codebase, so better suggestions are extremely welcome.